test/unit/model/navigation/LtiNavigationServiceTest.php

```diff
@@ -47,40 +60,223 @@ class LtiNavigationServiceTest extends TestCase
+    }
+
+    /**
+     * @param bool $thankYouScreenOption
+     * @param string $expectedUrl
+     *
+     * @dataProvider dataProviderTestGetReturnUrlNoSkipThankYouParameter
+     */
+    public function testGetReturnUrlNoSkipThankYouParameter(
+        bool $thankYouScreenOption,
+        string $ltiReturnUrl,
+        string $expectedUrl
+    ): void {
+        $this->mockLtiReturnUrlParameter($ltiReturnUrl);
+
+        $this->launchDataMock->method('hasVariable')
+            ->with('custom_skip_thankyou')
+            ->willReturn(false);
+
+        $this->object->setOption('thankyouScreen', $thankYouScreenOption);
+        $result = $this->object->getReturnUrl($this->launchDataMock, $this->deliveryExecutionMock);
+
+        static::assertSame($expectedUrl, $result, 'Method must return correct "return url"');
+    }
+
+    /**
+     * @param string $skipThankYou
+     * @param string $expectedUrl
+     *
+     * @dataProvider dataProviderTestGetReturnUrlSkipThankYouParameterValidValue
+     */
+    public function testGetReturnUrlSkipThankYouParameterValidValue(string $skipThankYou, string $expectedUrl): void
+    {
+        $this->mockLtiReturnUrlParameter();
+
+        $this->launchDataMock->method('hasVariable')
+            ->with('custom_skip_thankyou')
+            ->willReturn(true);
+        $this->launchDataMock->method('getVariable')
+            ->with('custom_skip_thankyou')
+            ->willReturn($skipThankYou);
+
+        $result = $this->object->getReturnUrl($this->launchDataMock, $this->deliveryExecutionMock);
+
+        static::assertSame($expectedUrl, $result, 'Method must return correct "return url"');
+    }
+
+    /**
+     * @param bool $thankYouScreenOption
+     * @param string $expectedUrl
+     *
+     * @dataProvider dataProviderTestGetReturnUrlSkipThankYouParameterInvalidType
+     */
+    public function testGetReturnUrlSkipThankYouParameterInvalidType(
+        bool $thankYouScreenOption,
+        string $expectedUrl
+    ): void {
+        $this->mockLtiReturnUrlParameter();
+
+        $skipThankYouLtiParameter = 'INVALID_VALUE';
+        $this->launchDataMock->method('hasVariable')
+            ->with('custom_skip_thankyou')
+            ->willReturn(false);
+        $this->launchDataMock->method('getVariable')
+            ->with('custom_skip_thankyou')
+            ->willReturn($skipThankYouLtiParameter);
+
+        $this->object->setOption('thankyouScreen', $thankYouScreenOption);
+        $result = $this->object->getReturnUrl($this->launchDataMock, $this->deliveryExecutionMock);
+
+        static::assertSame($expectedUrl, $result, 'Method must return correct "return url"');
+    }
+
+    public function dataProviderTestGetReturnUrlNoSkipThankYouParameter(): array
+    {
+        return [
+            'Option show thank you screen true' => [
+                'thankYouScreenOption' => true,
+                'ltiReturnUrl' => 'http://FAKE_LTI_RETURN.URL',
```

Comment:
On my machine these tests are failing because `parse_url` can not identify `path` in this fake url. `Undefined index: path`

Reply:
Fixed